src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/ReferralController.kt

```diff
@@ -41,4 +43,21 @@ class ReferralController(private val repository: ReferralRepository) {
       ?.let { ResponseEntity.ok(it) }
       ?: ResponseEntity.notFound().build()
   }
+
+  @PatchMapping("/draft-referral/{id}")
+  fun patchDraftReferralByID(@PathVariable id: String, @RequestBody partialUpdate: Referral): ResponseEntity<Any> {
+    val uuid = try {
+      UUID.fromString(id)
+    } catch (e: IllegalArgumentException) {
+      return ResponseEntity.badRequest().body("malformed id")
+    }
+
+    return repository.findByIdOrNull(uuid)
+      ?.let {
+        it.completionDeadline = partialUpdate.completionDeadline
```

Comment:
👍 done in d4ee226

This made me think...

As a REST API user, I'd expect `PATCH /draft-referral/:uuid` to reset the field, if the `completionDeadline` is explicitly specified, i.e. when the payload is `{"completionDeadline": null}`.

Is the presence of a key in the request possible to detect with a `@RequestBody` mapping? It would be great to throw back a validation error if we don't accept `null` values, rather than skipping them.